extensions/amp-analytics/0.1/instrumentation.js

```diff
@@ -212,14 +213,21 @@ export class InstrumentationService {
    * @private
    */
   createVisibilityListener_(callback, config) {
-    if (config['visibilitySpec']) {
+    const spec = config['visibilitySpec'];
+    const selector = config['selector'] || (spec && spec['selector']);
+    const analyticsVariables = config[vars] || {};
```

Comment:
but we don't do that for click event. In click event, the config level variables are merged with dataset variables in amp-analytics.js.

I think what you want to do is merge the `analyticsVariables` var with `vars` before Line 231.